lib/pbio/drv/uart/uart_ev3_pru.c

```diff
@@ -81,16 +91,23 @@ struct circ_buf {
 
 #define NR_SUART (2)
 #define SUART_CNTX_SZ 512
-#define SUART_FIFO_TIMEOUT_DFLT 10 // 5
-#define SUART_FIFO_TIMEOUT_MIN 4
-#define SUART_FIFO_TIMEOUT_MAX 500
 #define BUFFER_SIZE (1024) /* Needs to be a 2 size */
 #define BUFFER_MASK (BUFFER_SIZE - 1)
 
 #define __suart_err(fmt, args ...)
 
-/* Default timeout set to 5ms */
-static int32_t suart_timeout = SUART_FIFO_TIMEOUT_DFLT;
+static inline int suart_oversampling_rate_to_num(int x) {
+    if (x == SUART_8X_OVRSMPL) {
+        return 8;
+    }
+    if (x == SUART_16X_OVRSMPL) {
+        return 16;
+    }
+    assert(false);
+    // This is more conservative -- we will get the interrupts less frequently,
+    // but everything should still work.
+    return 16;
```

Comment:
```suggestion
    switch (x) {
    case SUART_8X_OVRSMPL:
        return 8;
    default:
        return 16;
    }
```

Could make it a bit more compact.